<a href="https://colab.research.google.com/github/fralfaro/MA_M3/blob/main/evaluaciones/tarea_02_visualizacion_eda.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Tarea 2: Visualizacion de datos y analisis exploratorio

## Objetivo

Desarrollar un analisis exploratorio visual de ventas e-commerce usando **Matplotlib**. El objetivo es transformar los KPI y tablas de la Tarea 1 en visualizaciones claras que permitan comunicar hallazgos y apoyar decisiones comerciales.

## Contexto

Se utilizara el mismo archivo `evaluaciones/data.csv`. Puede reutilizar el proceso de limpieza de la Tarea 1, dejando el codigo ordenado al inicio del notebook.

## Entrega

Entregar este notebook completo, ejecutado y ordenado. Cada grafico debe estar construido con Matplotlib, tener titulo, ejes claros e interpretacion breve.


## 0. Carga de librerias y datos

Cargue los datos, limpielos y cree las variables necesarias para visualizar ventas, clientes, productos y tiempo.

In [11]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

plt.style.use("bmh")
plt.rcParams.update({
    "figure.figsize": (10, 5),
    "axes.titlesize": 14,
    "axes.labelsize": 11,
    "legend.frameon": True,
    "legend.framealpha": 0.9,
})

ruta = "https://raw.githubusercontent.com/fralfaro/MA_M3/refs/heads/main/evaluaciones/data.csv"  # Si ejecutas desde Colab
# ruta = "evaluaciones/data.csv"  # Si ejecutas desde la raiz del repositorio
# ruta = "data.csv"  # Si ejecutas desde la carpeta evaluaciones

df = pd.read_csv(ruta, encoding="latin1")
df.head()


,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,12/1/2010 8:26,2.55,17850.0,United Kingdom
1,536365,71053,WHITE METAL LANTERN,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,12/1/2010 8:26,2.75,17850.0,United Kingdom
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,12/1/2010 8:26,3.39,17850.0,United Kingdom


In [12]:
# Limpieza base sugerida
df["InvoiceDate"] = pd.to_datetime(df["InvoiceDate"])
df["is_cancelled"] = df["InvoiceNo"].astype(str).str.startswith("C")

df_clean = df.drop_duplicates().copy()
df_clean = df_clean[
    (~df_clean["is_cancelled"]) &
    (df_clean["Quantity"] > 0) &
    (df_clean["UnitPrice"] > 0)
].copy()

df_clean["TotalAmount"] = df_clean["Quantity"] * df_clean["UnitPrice"]
df_clean["Year"] = df_clean["InvoiceDate"].dt.year
df_clean["Month"] = df_clean["InvoiceDate"].dt.month
df_clean["YearMonth"] = df_clean["InvoiceDate"].dt.to_period("M").astype(str)
df_clean["InvoiceDateOnly"] = df_clean["InvoiceDate"].dt.date

df_clean.head()

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country,is_cancelled,TotalAmount,Year,Month,YearMonth,InvoiceDateOnly
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,2010-12-01 08:26:00,2.55,17850.0,United Kingdom,False,15.30,2010,12,2010-12,2010-12-01
1,536365,71053,WHITE METAL LANTERN,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom,False,20.34,2010,12,2010-12,2010-12-01
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,2010-12-01 08:26:00,2.75,17850.0,United Kingdom,False,22.00,2010,12,2010-12,2010-12-01
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom,False,20.34,2010,12,2010-12,2010-12-01
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom,False,20.34,2010,12,2010-12,2010-12-01


## Preguntas

### 1. Evolucion mensual de ventas

Construya un grafico de linea con la venta total por mes.

Responda:

- En que meses se observan mayores ventas?
- Existe algun patron temporal relevante?
- Que recomendacion comercial se podria proponer a partir del grafico?

### 2. Ranking de paises por venta

Construya un grafico de barras horizontales con los 10 paises con mayor venta total.

Responda:

- Que pais concentra la mayor venta?
- Conviene incluir o excluir United Kingdom para observar mejor el resto de paises? Justifique.
- Que oportunidad comercial podria existir en otros mercados?

### 3. Productos con mayor venta

Construya un grafico de barras con los 10 productos de mayor venta total.

Responda:

- Que productos lideran la venta?
- El ranking parece depender mas de volumen, precio o ambos?
- Que decision de inventario o promocion se podria tomar?

### 4. Distribucion del ticket por factura

Calcule el monto total por factura y visualice su distribucion usando histograma o boxplot.

Responda:

- Como se distribuyen los tickets de compra?
- Existen valores extremos?
- Que implicancia tiene esto para interpretar promedios?

### 5. Clientes de mayor valor

Identifique los 10 clientes con mayor venta total y visualicelos con un grafico de barras.

Responda:

- Que tan concentrada esta la venta en los principales clientes?
- Que accion de fidelizacion podria recomendar?
- Que limitacion tiene este analisis si existen `CustomerID` nulos?

### 6. Relacion entre cantidad, precio y venta

Construya un grafico de dispersion entre `Quantity` y `UnitPrice`, usando `TotalAmount` como referencia adicional si lo considera necesario.

Responda:

- Que relacion observa entre cantidad y precio unitario?
- Existen productos de alto volumen y bajo precio?
- Existen productos de bajo volumen y alto precio?
- Que precauciones debe tomar al graficar valores extremos?

### 7. Comparacion de indicadores por pais

Para los principales paises, compare al menos dos indicadores en una visualizacion:

- Venta total.
- Numero de facturas.
- Ticket promedio.
- Clientes unicos.

Puede usar barras agrupadas, subplots o una tabla visual complementada con grafico.

### 8. Storytelling con datos

Seleccione los 3 graficos mas importantes del notebook y redacte una conclusion ejecutiva de 8 a 12 lineas. La conclusion debe responder:

- Que esta pasando con las ventas?
- Donde estan las principales oportunidades?
- Que riesgos o limitaciones tienen los datos?
- Que accion concreta recomendaria al equipo de marketing o ventas?

## Rubrica sugerida

| Criterio | Ponderacion |
|---|---:|
| Limpieza y preparacion del dataset | 15% |
| Calidad tecnica de los graficos | 30% |
| Variedad y pertinencia de visualizaciones | 20% |
| Interpretacion orientada a negocio | 25% |
| Orden, reproducibilidad y presentacion | 10% |